pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -715,17 +715,13 @@ func (c *Cluster) makeDeployment(osdProps osdProperties, osd OSDInfo, provisionC
 	}
 
 	if osdProps.onPVC() {
-		// the "all" placement is applied separately so it will have lower priority.
-		// We want placement from the storageClassDeviceSet to be applied and override
-		// the "all" placement if there are any overlapping placement settings.
-		c.spec.Placement.All().ApplyToPodSpec(&deployment.Spec.Template.Spec)
-		// apply storageClassDeviceSet Placement
-		// If nodeAffinity is specified both in the device set and "all" placement,
-		// they will be merged.
+		c.placementForOsdandPrepareOsdPods(&deployment.Spec.Template.Spec)
+		// apply storageClassDeviceSets.Placement
 		osdProps.placement.ApplyToPodSpec(&deployment.Spec.Template.Spec)
 	} else {
-		p := cephv1.GetOSDPlacement(c.spec.Placement)
-		p.ApplyToPodSpec(&deployment.Spec.Template.Spec)
+		c.placementForOsdandPrepareOsdPods(&deployment.Spec.Template.Spec)
```

Comment:
How about taking the argument for applying the placement after the `all` placement?
```suggestion
		c.placementForOsdandPrepareOsdPods(c.spec.Placement[cephv1.KeyOSD], &deployment.Spec.Template.Spec)
```

Reply:
not changing as types for PVCs OSD and non-PVCs OSD are different